src/swish/osi.c

```diff
@@ -888,6 +888,54 @@ ptr osi_spawn(const char* path, ptr args, ptr callback) {
   return r;
 }
 
+ptr osi_spawn_detached(const char* path, ptr args) {
+  int argc = string_list_length(args);
+  if (argc < 0)
+    return osi_make_error_pair("osi_spawn_detached", UV_EINVAL);
+  // Build argument list
+  char** argv = (char**)malloc((argc + 2) * sizeof(char*));
+  if (!argv)
+    return osi_make_error_pair("osi_spawn_detached", UV_ENOMEM);
+  {
+    char** arg = argv;
+    *arg++ = (char*)path;
+    ptr x = args;
+    while (Spairp(x)) {
+      size_t l;
+      if (!(*arg++ = osi_string_to_utf8(Scar(x), &l)))
+        break;
+      x = Scdr(x);
+    }
+    *arg = NULL;
+  }
+  // Spawn the process
+  uv_process_t* p = malloc_container(uv_process_t);
+  if (!p) {
+    free_argv(argv);
+    return osi_make_error_pair("osi_spawn_detached", UV_ENOMEM);
+  }
+  uv_process_options_t options = {
+    .exit_cb = NULL,
+    .file = path,
+    .args = argv,
+    .env = NULL,
+    .cwd = NULL,
+    .flags = UV_PROCESS_DETACHED,
+    .stdio_count = 0,
+    .stdio = NULL,
+    .uid = 0,
+    .gid = 0
+  };
+  p->data = 0;
+  int rc = uv_spawn(osi_loop, p, &options);
+  free_argv(argv);
+  if (rc < 0) {
+    uv_close((uv_handle_t*)p, NULL);
+    return osi_make_error_pair("uv_spawn", rc);
+  }
+  return Sinteger32(p->pid);
```

Comment:
This leaks the uv_process structure and doesn't close it. We need to define an exit_cb that mimics process_exit_cb but without the callback.

Reply:
Upon further reading, I don't think we want to hook an exit_cb, as that is fired when the process exits. By definition, we don't want to be monitoring the process. I do however see in the LibUV docs that I should've called `uv_unref`. Maybe I can call `uv_close` directly here. See what you think of the patch.

Reply:
We still need to free the memory for p.

Reply:
I tried to use an automatic variable for the uv_process_t and avoid the malloc & free altogether. Unfortunately, uv_close threw an invalid memory reference when I did.

Reply:
I pushed commits to simplify the closing process, and it appears to work on the Mac. I'll try it on Windows shortly.

Reply:
On Unix systems, the child is still watched even after we uv_close. I'm going to propose an exit_cb because it cleans everything up correctly.

Reply:
Even with the latest commit, I get an invalid memory reference when I hit CTRL-C on the Mac from a loop that continuously spawns "sync". Something's still fishy here.

Reply:
I found that uv_close always asynchronously calls its callback, even when it's NULL. This means we shouldn't call it with NULL and then immediately free the data structure. See the latest commit.